packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -130,6 +136,10 @@ class Dismissible extends StatefulWidget {
   final DismissDirectionCallback onDismissed;
 
   /// The direction in which the widget can be dismissed.
+  ///
+  /// If null, the widget will not be dismissible.
+  ///
+  /// The widget's [child] should not be null if [direction] is null.
```

Comment:
should -> must? We assert if you set both to null.

Reply:
It's also a strange limitation.

Reply:
Yeah, from the original PR discussion, re line #526 below
> If widget.child is null, performRebuild will throw. Could you assert in the constructor that child and direction can't both be null?